src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -172,6 +185,14 @@ public void AddConventionalLinkGenerationRoute(List<Endpoint> endpoints, HashSet
                 builder.Metadata.Add(new RouteNameMetadata(route.RouteName));
             }
 
+            // MVC guarantees that when two of it's endpoints have the same route name they are equivalent.
+            //
+            // However, Endpoint Routing requires Endpoint Names to be unique.
+            if (route.RouteName != null && routeNames.Add(route.RouteName))
```

Comment:
routeNames.Add will return true if the add inserted a new route name, otherwise false.

What is a scenario that will return false? I'd like to understand the experience.

Is it worth debug logging when false is returned?

Reply:
The scenario is the one in the tests and the discussed here

```C#
[HttpGet]
[HttpPost]
[Route(... Name = "Foo")]
public void DoStuff() { }
```

This creates multiple endpoints but they are equivalent for URL generation. The other alternative is to totally block this feature. I don't really feel like it's worth having logging for this case because nothing's broken and it's fairly common. 

The opposite case (different actions with same route name) is already an error.